### ISSUE TITLE
FlareSolverr 3 : Plus de session

### ISSUE THREAD

**Issue by simpledork47** [Kodi-vStream/venom-xbmc-addons#3999]
Depuis la sortie de FlareSolverr 3, les méthodes "sessions" ne sont plus disponible = "deprecated"

https://github.com/FlareSolverr/FlareSolverr/blob/2d97f882764caf6563647109871b5230d5899840/src/flaresolverr_service.py#L117

ça doit être possible d'enrober le code existant dans du try/except qui puisse marcher sur un FlareSolverr plus ancien j'imagine

**Comment by GravuTrad** [comment]
Le plus ancien ne passe plus cloudflare pour ma part, donc à mon avis on ne peut que attendre que ngosang réimplémente ces éléments dans sa version 3 (c'est dans sa todo list me semble t'il)

**Comment by simpledork47** [comment]
j'ai fais une pull request
ici c'est nickel avec 'VoirAnime'

**Comment by GravuTrad** [comment]
Je ne vois plus le pull il est où?

**Comment by sizanic** [comment]
Il est dans le code de la version.

vStream V2.4.3, compatible FlareSolverr 3 : 
https://github.com/Kodi-vStream/venom-xbmc-addons/blob/Beta/plugin.video.vstream/changelog.txt

**Comment by GravuTrad** [comment]
Merci sizanic je vais tester l'évolution

**Comment by TmpName** [comment]
Il va juste manquer la partie pour les requetes en POST, mais c'est vite rajouté, le seul truc c'est que je n'ai pas de site pour tester. Donc tiens nous au courant si ca bloque sur un site precie.

**Comment by GravuTrad** [comment]
le souci est que tant que le bug de l'erreur 500 de flaresolverv3 ne sera pas résolu cela bloqera pour moi sur tous les sites. Mais les requetes passent bien, juste que flaresolverr nous sort une erreur serveur 500
https://github.com/FlareSolverr/FlareSolverr/issues/668
erreur 500:
https://github.com/FlareSolverr/FlareSolverr/issues/667

**Comment by TmpName** [comment]
Ha y a un nouveau soucis ?

Du coup flaresolver marche bien, mais ne retourne plus la page ?

**Comment by GravuTrad** [comment]
je vais essayer la version windows. Car c'est la version docker sur serveur web qui donne une erreur 500

**Comment by GravuTrad** [comment]
Perso j'ai basculé sur celui ci:
https://github.com/smeinecke/FlareSolverr
plus d'updates que l'officiel et plus fonctionnel et réactif.

**Comment by TmpName** [comment]
Il y a une nouvelle lib dans vstream qui marche nativement dans certains cas, mais elle n'est pas reglée au max par defaut
https://github.com/Kodi-vStream/venom-xbmc-addons/tree/Beta/plugin.video.vstream/resources/lib/cloudscraper

**Comment by sizanic** [comment]
merci @GravuTrad 
j'ai  référencé cette version dans notre doc : 
https://github.com/Kodi-vStream/venom-xbmc-addons/wiki/Comment-passer-CloudFlare

**Comment by GravuTrad** [comment]
pour l'instant cpasmal ca passe pas. même avec cloudscraper.
et de toute façon flaresolverr en ce moment ca me met de ça:
"Error: Error solving the challenge. Cloudflare has blocked this request. Probably your IP is banned for this site, check in your web browser", des challenge not solved et des erreurs 500...
et je pense que avec vpn ca passe.

**Comment by GravuTrad** [comment]
pour cloudscraper vous utilisez quelle version?

**Comment by TmpName** [comment]
La version incluse dans Vstream, c'est un fork, pas la version d'origine.

**Comment by GravuTrad** [comment]
Bien celui ci? https://github.com/zinzied/cloudscraper

**Comment by sizanic** [comment]
non, il me semble que c'est : https://github.com/lncrawl/cloudscraper

**Comment by GravuTrad** [comment]
Ah. Parce que vous avez vu que même cette page, comme sur l'officiel, ils indiquent tous les deux que la version la plus poussée est celle de zinzied.
Le coup d'utiliser gemini comme solver m'épate lol

**Comment by sizanic** [comment]
je ne sais pas si c'est cette version, j'ai bien vu des versions plus récentes, mais "trop" récente 😄 
Par rapport à la version de Python.
J'ai préféré misé sur la compatibilité des versions avec du code python plus ancien.
